src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/full_screen_waterfall/hooks/use_fetch_log/index.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import { lastValueFrom } from 'rxjs';
+import { SPAN_ID_FIELD, TRACE_ID_FIELD, TRANSACTION_ID_FIELD } from '@kbn/discover-utils';
+
+import { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { useDataSourcesContext } from '../../../../hooks/use_data_sources';
+import { getUnifiedDocViewerServices } from '../../../../../../../plugin';
+
+interface Props {
+  traceId: string;
+  docId: string;
+}
+
+interface GetTransactionParams {
+  traceId: string;
+  docId: string;
+  indexPattern: string;
+  data: DataPublicPluginStart;
+  signal: AbortSignal;
+}
+
+async function fetchLogDocument({
+  traceId,
+  docId,
+  indexPattern,
+  data,
+  signal,
+}: GetTransactionParams) {
+  return lastValueFrom(
+    data.search.search(
+      {
```

Comment:
I've noticed we're now using lastValueFrom(data.search.search(...)) for span, root trace and we keep adapting it. 
What are the key benefits of using this approach? Also, how can we write tests around it?